presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -145,17 +146,22 @@ private ExtendedFileSystem getFileSystem(ConnectorSession session, HudiTableHand
 
     private HudiSplit createHudiSplit(HudiTableHandle table, FileSlice slice, String timestamp, HudiPartition partition)
```

Comment:
A better way to handle split generation:

- change return type of createHudiSplit to `Optional<HudiSplit>`
-  if `table` type is COW and `slice` has no base file, return an empty `Optinal<HudiSplit>`
- filter empty optionals before `builder::add` in Line125.
- `HiveEmptySplitPageSource` is not needed any longer.

This might make more sense. Because we don't need to dispatch an empty split.